internal/hcp/internal/controllers/link/controller.go

```diff
@@ -123,7 +120,14 @@ func (r *linkReconciler) Reconcile(ctx context.Context, rt controller.Runtime, r
 		return writeStatusIfNotEqual(ctx, rt, res, newStatus)
 	}
 
-	hcpClient, err := r.hcpClientFn(&link)
+	existingCfg := r.hcpManager.GetCloudConfig()
+	newCfg := config.CloudConfig{
+		ResourceID:   link.ResourceId,
+		ClientID:     link.ClientId,
+		ClientSecret: link.ClientSecret,
+	}
+	cfg := config.Merge(existingCfg, newCfg)
+	hcpClient, err := r.hcpClientFn(cfg)
```

Comment:
I think this makes sense, but just FYI, we'll have a merge conflict with Nick E's change: https://github.com/hashicorp/consul/pull/20257/files

Reply:
Thanks for the heads up, will keep this in mind!